### PR TITLE
[Performance][KDA] Compose and overlap gate projections on main

### PR DESCRIPTION
**Pull request by Dawn952**
### What this PR does / why we need it?

This ports #14497 onto main after the Kimi K3 integration in #14454.

For the existing mixed-precision Kimi K3 KDA layout, this change:

- composes f_proj.weight = f_b_proj.weight @ f_a_proj.weight after checkpoint loading and later source-weight reloads;
- packs beta, the composed F projection, and the output gate into one floating-point BFG projection;
- runs a two-stage auxiliary-stream schedule:
  1. main-stream MXFP DynamicQuant overlaps the auxiliary BFG GEMM;
  2. main-stream QKV GEMM overlaps auxiliary B/F/G split, beta FP32 sigmoid, and gate reshaping;
- marks mixed-path beta as preprocessed so the v0.27 dispatch path only slices it, while ordinary upstream raw beta retains its existing FP32 sigmoid path;
- preserves the configured W8A8 MXFP8 scale_alg when DynamicQuant is split from the linear method;
- routes global or local F shards through the v0.27 packed loader;
- leaves the ordinary upstream in_proj_qkvgfab path unchanged;
- removes legacy next-power-of-two MLA head padding now that the replacement attention operator accepts arbitrary head counts.

This replaces the now-conflicting #15124, which was opened before #14454 merged and therefore also contained the full Kimi K3 integration.

### ACL graph and stream-order assessment

The auxiliary stream records bfg_ready as its final node, and the main stream waits for that exact tail event. No auxiliary operation is queued after the join point.

The four-event schedule is intentional:

1. hidden_states_ready forks BFG from main;
2. bfg_projection_ready serializes the BFG and QKV Cube matmuls;
3. quant_ready starts auxiliary beta vector work only after main DynamicQuant;
4. bfg_ready joins the complete auxiliary tail before KDA consumes its outputs.

The live-token slices for mixed_qkv, g1, and g2 remain at the eager dispatch boundary. They are host-side basic views: after QKV GEMM and the auxiliary tail wait are enqueued, Python creates those views while the asynchronous device work is still running. Moving num_actual_tokens into the projection path would not add an NPU kernel overlap and would weaken the runtime metadata boundary.

Static assessment: the stream is fully joined for ACLGraph capture and mixed beta is transformed exactly once. A real NPU ACLGraph capture/replay has not yet been run for this main port.

### How was this patch tested?

- python -m ruff format --check on all six changed files.
- python -m ruff check on all six changed files.
- Python syntax compilation on all six changed files.
- git diff --check.
- Added focused CPU tests for:
  - global/local F shard composition and reload;
  - BFG projection versus delayed split/post-processing;
  - raw beta and auxiliary-preprocessed beta semantics;
  - mixed forward wiring for preprocessed beta;
  - two-stage stream event ordering and graph-safe auxiliary-tail join;
  - MXFP DynamicQuant split and W8A8 MXFP8 scale_alg propagation;
  - non-MXFP quantization fallback;
  - v0.27 packed checkpoint routing;
  - non-power-of-two MLA head counts remaining unpadded.
- Replaced append-expression lambdas in the event-order test with typed helpers to satisfy repository mypy.

The focused pytest files could not be collected on the Windows workstation because its Python environment does not include vllm. No service, inference request, or benchmark was run for this port.

### Does this PR introduce any user-facing change?

No API or configuration change. Runtime behavior changes only for Kimi K3 KDA layers using the existing mixed-precision full-rank gate layout.

Related: #13727, #14454, #14497, #15124

- vLLM version: v0.27.1
- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

File: vllm_ascend/ops/kimi_kda.py [hunk 1]
```diff
@@ -180,6 +278,99 @@ def forward(
...
+            bfg_stream.wait_event(quant_ready)
+            beta, g1, g2 = self._postprocess_bfg(*raw_bfg)
+            bfg_ready = bfg_stream.record_event()
+
+        for tensor in (beta, g1, g2):
+            tensor.record_stream(main_stream)
+        # bfg_ready is the auxiliary stream tail. Joining that exact event is
+        # required for multi-stream ACL graph capture as well as eager reuse.
+        main_stream.wait_event(bfg_ready)
+        return mixed_qkv, beta, g1, g2
```
[the remark this answers is not in the document]
Comment (Dawn952) [reply vllm_ascend/ops/kimi_kda.py:322]: Addressed in fd8125e5 with the original two-stage resource schedule restored rather than the proposed Cube/Cube overlap. The auxiliary second stage now performs the fused B/F/G split plus beta.float().sigmoid() after the main-stream QKV GEMM is enqueued, so quant_ready guards real vector work again. We retain bfg_projection_ready to serialize the two Cube matmuls and bfg_ready remains the final joined auxiliary node for ACLGraph capture. The event-order and raw/preprocessed-beta tests were updated accordingly.

File: vllm_ascend/ops/kimi_kda.py [hunk 2]
```diff
@@ -29,9 +29,125 @@
...
+    @torch.no_grad()
+    def _maybe_fuse_f_proj(self) -> None:
+        if not self._f_a_loaded or not self._f_b_loaded:
+            return
+        output_dim = getattr(self.weight, "output_dim", None)
+        if output_dim is None:
+            raise ValueError("KDA fused f_proj requires an output-sharded parameter")
+        shard_offset = sum(self.output_sizes[:_F_PROJ_SHARD_ID]) // self.tp_size
+        shard_size = self.output_sizes[_F_PROJ_SHARD_ID] // self.tp_size
+        param_shard = self.weight.data.narrow(output_dim, shard_offset, shard_size)
```
[the remark this answers is not in the document]
Comment (Dawn952) [reply vllm_ascend/ops/kimi_kda.py]: Addressed in fd8125e5. _maybe_fuse_f_proj is already under torch.no_grad(), so the shard view now uses self.weight.narrow(...) directly.
